packages/tizen-remote/lib/index.js

```diff
@@ -289,12 +352,105 @@ export class TizenRemote extends createdTypedEmitterClass() {
   }
 
   /**
-   * Computed URL of the Tizen web socket server.
+   * Just returns the client name (which is a base64-encoded string).
+   */
+  get base64Name() {
+    return this.#name;
+  }
+
+  /**
+   * Computed URL of the Tizen device's websocket endpoint (as a string)
    */
   get url() {
     return this.#url.toString();
   }
 
+  /**
+   * The key to use for this client in the token cache
+   */
+  get #tokenCacheKey() {
+    return `${this.#name}.token`;
+  }
+
+  /**
+   * Unsets token.
+   *
+   * If token cache persistence is enabled, this will remove it from the cache as well.
+   */
+  async unsetToken() {
+    if (this.#persistToken) {
+      const cache = await this.#getTokenCache();
+      await this.#lock();
+      cache.delete(this.#tokenCacheKey);
+      await this.#unlock();
+    }
+    this.#token = undefined;
+    this.#debug('Reset token for client %s', this.#name);
+  }
+
+  /**
+   * Locks the lockfile
+   *
+   * Creates the cache dir if it does not exist.
+   */
+  async #lock() {
+    await fs.mkdir(this.#cacheDir, {recursive: true});
+    await lock(this.#lockfilePath);
+  }
+
+  /**
+   * Unlocks the lockfile
+   */
+  async #unlock() {
+    await fs.mkdir(this.#cacheDir, {recursive: true});
+    await unlock(this.#lockfilePath);
+  }
+
+  /**
+   * Initializes the token cache; otherwise returns the existing cache.
+   *
+   * Because the `Conf` constructor performs a blocking write to the fs, we need to
+   * wrap the instantiation in a lock.
+   *
+   * Aditionally, any future writes to the cache must be wrapped in the lock.
+   * @returns {Promise<Conf<Record<string,string>>>}
+   */
+  async #getTokenCache() {
+    /** @type {Conf<Record<string,string>>} */
+    let cache;
```

Comment:
seems like we don't need a local variable, could just be:

```js
if (!this.#tokenCache) {
  this.#tokenCache = ...
}
return this.#tokenCache
```

Reply:
yeah, this is just getting carried away with my coding style